File: programmers/high_score_kit/해시/위장.py
def solution(clothes):
    answer = 1
    cloth = dict()

    for i in range(len(clothes)):
        if clothes[i][1] not in cloth.keys():
            cloth[clothes[i][1]] = 0
        cloth[clothes[i][1]] += 1

    for i in cloth.keys():
        answer *= cloth[i] + 1

    answer = answer - 1

    return answer

print(solution([["yellowhat", "headgear"], ["bluesunglasses", "eyewear"], ["green_turban", "headgear"]]))
print(solution([["crowmask", "face"], ["bluesunglasses", "face"], ["smoky_makeup", "face"]]))

File: programmers/high_score_kit/해시/위장.js
function solution(clothes) {
    let answer = 1;
    // clothes 종류를 key로, 가짓수를 value로
    let cloth = {};
    clothes.map((el) => cloth[el[1]] = 0);
    clothes.map((el) => cloth[el[1]] += 1);

    Object.values(cloth).map((el) => answer *= (el + 1)); // 각각 종류 안입는 경우 포함해서 경우의 수 계산한다.
    answer -= 1; // 아무것도 안 입는 경우 빼준다
    return answer;
}

console.log(solution([["yellowhat", "headgear"], ["bluesunglasses", "eyewear"], ["green_turban", "headgear"]])); // 5